sql/mysql/internal/mysqlcheck/datadepend.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2021-present The Atlas Authors. All rights reserved.
+// This source code is licensed under the Apache 2.0 license found
+// in the LICENSE file in the root directory of this source tree.
+
+package mysqlcheck
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"ariga.io/atlas/sql/mysql"
+
+	"ariga.io/atlas/sql/schema"
+
+	"ariga.io/atlas/schema/schemaspec"
+	"ariga.io/atlas/sql/sqlcheck"
+	"ariga.io/atlas/sql/sqlcheck/datadepend"
+)
+
+type (
+	// DataDependOptions defines the configuration options
+	// for the data-dependent changes checker.
+	DataDependOptions struct {
+		// NotNull indicates if the analyzer should check for modification or
+		// addition of not null constraints to
+		NotNull *bool `spec:"not_null,omitempty"`
+
+		// Allow drivers to extend the configuration.
+		schemaspec.DefaultExtension
+	}
+
+	// DataDepend checks data-dependent changes for MySQL.
+	DataDepend struct {
+		DataDependOptions
+		*datadepend.Analyzer
+	}
+)
+
+// Analyze runs data-depend analysis on MySQL changes.
+func (d *DataDepend) Analyze(ctx context.Context, p *sqlcheck.Pass) error {
+	diags := d.Diagnostics(ctx, p)
+	for _, sc := range p.File.Changes {
+		for _, c := range sc.Changes {
+			m, ok := c.(*schema.ModifyTable)
+			if !ok {
+				continue
+			}
+			for _, c := range m.Changes {
+				switch c := c.(type) {
```

Comment:
Keeping it as `switch` and not `if` because in the next PRs I'll add support for column modifications (nullability, charset, etc).